Test the version number for semver compatibility

bdudelsack opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/master/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

Adds a test for semver compatibility

closes: #8645 

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

mattfarina commented on internal/version/version_test.go:
```diff
@@ -0,0 +1,31 @@
+)
+
+func TestVersionSemver(t *testing.T) {
+	_, err := semver.Parse(version)
```
Some context... while a version of Helm is being developed the version is set to the minor version only. You end up with versions output by the `version` command of `3.3+unreleased`.

When Helm is compiled `helm.sh/helm/v3/internal/version.version` is set as part of the processes with a semantic version. The `Makefile` does this for us.

mattfarina reviewed (changes requested):
The test here fails due to the internal version not being SemVer. That would need to be fixed, so the version passes, as part of the PR. We can't merge failing tests in.

mattfarina commented:
@bdudelsack Helm uses a [DCO](https://developercertificate.org/). Can you please add a Git signoff. The _checks_ tab on the PR has directions on fixing the commit.

bdudelsack committed:
Test the version number for semver compatibility

Signed-off-by: Boris Dudelsack <bdudelsack@gmail.com>

bdudelsack commented:
Is it intented to use a version without the patch number in development? It's not clear for me if i'm on the right way doing this PR here or is it a problem of package creators?

I've corrected the Version to v3.3.0, added a test for semver with Masterminds/semver packages and corrected other tests to comply with the new version.

bacongobbler commented on pkg/lint/rules/chartfile.go:
```diff
@@ -22,11 +22,12 @@ import (
 	"github.com/pkg/errors"
 	"sigs.k8s.io/yaml"
 
+	"github.com/Masterminds/semver/v3"
```
What's with the move here? Can you move this back?

bdudelsack replied:
It was moved by the `gitimports` command. I reverted the changes.

bacongobbler reviewed (changes requested):
Can you please move the import statements back to their original location?

The reason we never added the patch version number was because we did not bump the internal version during patch numbers.

Unless we update the internal version every patch release, this is going to cause odd scenarios where the development version is `v3.3.0+unreleased`, but the latest release is `v3.3.2`, which is why we chose to use `v3.3+unreleased` in the first place.

@adamreese and others may have more opinions here on this change.
